.clj-kondo/config.edn

```diff
@@ -1,10 +1,24 @@
 {:linters
-  {:unresolved-symbol
-   {:exclude [(puppetlabs.trapperkeeper.testutils.bootstrap/with-app-with-config [app])]}}
- :lint-as {puppetlabs.trapperkeeper.core/defservice clj-kondo.lint-as/def-catch-all
-           ctia.lib.compojure.api.core/GET clj-kondo.lint-as/def-catch-all
-           ctia.lib.compojure.api.core/PATCH clj-kondo.lint-as/def-catch-all
-           ctia.lib.compojure.api.core/POST clj-kondo.lint-as/def-catch-all
-           ctia.lib.compojure.api.core/PUT clj-kondo.lint-as/def-catch-all
-           ctia.lib.compojure.api.core/DELETE clj-kondo.lint-as/def-catch-all
-           ctia.lib.compojure.api.core/context clj-kondo.lint-as/def-catch-all}}
+ {:skip-comments true
+  :unresolved-symbol
+  {:exclude [(puppetlabs.trapperkeeper.testutils.bootstrap/with-app-with-config [app])]}
+  :unresolved-var
+  {:exclude [clj-momo.lib.clj-time.core/seconds]}}
+ :lint-as {puppetlabs.trapperkeeper.core/defservice  clj-kondo.lint-as/def-catch-all
+           ctia.lib.compojure.api.core/GET           clj-kondo.lint-as/def-catch-all
+           ctia.lib.compojure.api.core/PATCH         clj-kondo.lint-as/def-catch-all
+           ctia.lib.compojure.api.core/POST          clj-kondo.lint-as/def-catch-all
+           ctia.lib.compojure.api.core/PUT           clj-kondo.lint-as/def-catch-all
+           ctia.lib.compojure.api.core/DELETE        clj-kondo.lint-as/def-catch-all
+           ctia.lib.compojure.api.core/context       clj-kondo.lint-as/def-catch-all
+
+           ctia.schemas.core/defschema               clj-kondo.lint-as/def-catch-all
+           ctia.schemas.core/def-acl-schema          clj-kondo.lint-as/def-catch-all
+           ctia.schemas.core/def-stored-schema       clj-kondo.lint-as/def-catch-all
+           ctia.schemas.core/def-advanced-acl-schema clj-kondo.lint-as/def-catch-all
+
+           ctia.stores.es.store/def-es-store         clojure.core/defrecord
```

Comment:
@frenchy64 Perhaps you can help me with this.
The problem with that macro `ctia.stores.es.store/def-es-store`, that it creates a defrecord.
But when I force kondo to lint it as one, it then chokes on other syntactic elements, i.e., 

```lisp
(def-es-store AssetStore :asset StoredAsset PartialStoredAsset)
```

would complain: "unsupported binding form :asset"

But if I do `clj-kondo.lint-as/def-catch-all`, it then doesn't recognize `->AssetStore`, that defrecord declares.

Do you know a good way of dealing with this macro? Maybe using hooks (which I don't understand well)

Reply:
Try creating a file `.clj-kondo/ctia/stores/es/store.clj` that looks something like:

```clojure
(ns ctia.stores.es.store)
(defmacro def-es-store
  [store-name
   entity
   stored-schema
   partial-stored-schema]
  `(clojure.core/defrecord ~store-name [~(symbol "state")]))
```

Then add this entry to `.clj-kondo/config.edn` (and remove other mentions of `ctia.stores.es.store/def-es-store`).

```clojure
{:hooks {:macroexpand {ctia.stores.es.store/def-es-store ctia.stores.es.store/def-es-store}}}
```